src/components/List/ListItem.tsx

```diff
@@ -1,20 +1,30 @@
-import type { FC, PropsWithChildren } from 'react';
+import type { ComponentProps, FC, PropsWithChildren } from 'react';
 import { twMerge } from 'tailwind-merge';
 import { mergeDeep } from '../../helpers/merge-deep';
 import { getTheme } from '../../theme-store';
 import type { DeepPartial } from '../../types';
 
 export interface FlowbiteListItemTheme {
-  base: string;
+  icon: string;
+  withIcon: {
+    on: string;
+    off: string;
+  };
 }
 
 export interface ListItemProps extends PropsWithChildren {
```

Comment:
i'd rather have the API more elastic, giving it `ComponentProps<'li'>` and spreading the `{...props}` on the `<li>` component

Reply:
instead of `PropsWithChildren` interface u could do `ComponentProps<'li'>` to allow all vanilla `li` HTML element props to be passed to this React list component

Reply:
Sounds awesome ;)
Done.